File: BST/CountBSTinRange/count.cpp
#include <vector>
using namespace std;

struct Node {
    int data;
    Node* left;
    Node* right;

    Node(int x){
        data = x;
        left = right = NULL;
    }
};

class Solution {
public:
    void prepareInVector(Node* root, vector<int>& ans) {
        if (!root)
            return;

        prepareInVector(root->left, ans);
        ans.push_back(root->data);
        prepareInVector(root->right, ans);
    }

    int getCount(Node* root, int l, int h) {
        vector<int> nodes;
        prepareInVector(root, nodes);

        int count = 0;

        for (int val : nodes) {
            if (l <= val && val <= h)
                count++;
        }

        return count;
    }
};

File: BST/CountBSTinRange/count.py
class Node:
    def __init__(self, val):
        self.right = None
        self.data = val
        self.left = None

class Solution:
    def prepareInVector(self, root, ans):
        if not root: return
        self.prepareInVector(root.left, ans)
        ans.append(root.data)
        self.prepareInVector(root.right, ans)
        
    def getCount(self, root, l, h):
        # Your code here
        count = 0
        nodes = []
        self.prepareInVector(root,nodes)
        n = len(nodes)
        for i in range(n):
            if l<=nodes[i] and nodes[i]<=h: count += 1
        return count